src/image/Exposure.cc

```diff
@@ -196,6 +198,72 @@ void Exposure<ImageT, MaskT, VarianceT>::writeFits(fits::Fits &fitsfile,
     _info->_finishWriteFits(fitsfile, data);
 }
 
+namespace {
+/**
+ * Create a box centered as closely as possible on a particular point.
+ *
+ * @param center The desired center of the box.
+ * @param size The desired width and height (in that order) of the box.
+ *
+ * @returns a box with size ``size`` and center within half a pixel of ``center`` in either dimension.
+ *
+ * @throws lsst::pex::exceptions::InvalidParameterError Thrown if ``size`` has invalid dimensions.
+ */
+lsst::geom::Box2I _makeCenteredBox(lsst::geom::Point2D const &center, lsst::geom::Extent2I const &size) {
```

Comment:
I think this would be a useful public addition to `Box2D`, either as a static member function or a free function in the same namespace.

Related methods that create `Image`, `Mask`, or `MaskedImage` cutouts given a non-integer center coordinate would be similarly useful, and with this function they'd be trivial.

Reply:
This goes back to the point you made by email about "conventions" (which apply to both even and odd dimensions, BTW: for an even size there are two or four equally good solutions if `center` lies on a pixel centerline, for an odd size there are multiple solutions if `center` lies on a pixel boundary). For the specific purpose of `getCutout`, it really doesn't matter which of four equally shifted bounding boxes is chosen (our application is source images for human consumption), and I was careful to neither promise nor test anything more specific.

A public method or function, on the other hand, may have use cases where the user does actually care which of the multiple solutions gets chosen (though, to be fair, I found in testing that floating-point error in the generation of `center` will almost always cause one of the four points to be slightly closer). It would need a more specific postcondition than what I've provided here, and the right answer might well depend on what the user wants to do with `center`.

Reply:
I'd be quite content having a public method that also documents that it does not make guarantees about which of these solutions are selected (and also does not guarantee that the behavior will not change in the future, if we want to start providing those guarantees).  I'd like to at least support the use cases that have the same (lenient) requirements as `getCutout`.

Reply:
I'm a little worried by the wording "does not guarantee that the behavior will not change in the future" -- it sounds like you'd want to make arbitrary changes, rather than merely making the output more specific. :+1: if I misunderstood.